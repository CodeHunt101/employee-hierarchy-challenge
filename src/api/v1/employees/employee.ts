export class Employee {
  readonly managerId?: number | null

  constructor(readonly name: string, readonly id: number, managerId?: number) {
    this.name = name
    this.id = id
    this.managerId = managerId ?? null
  }
}
