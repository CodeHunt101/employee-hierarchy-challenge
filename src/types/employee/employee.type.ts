import { Employee } from '../../api/v1/employees/employee'

export type EmployeeHierarchy = Employee & {
  depth: number
}

export type EmployeesHierarchyData = {
  employeesHierarchy: EmployeeHierarchy[]
  maxDepth: number
}
