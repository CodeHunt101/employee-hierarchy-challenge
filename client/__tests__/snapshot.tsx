import { render } from '@testing-library/react'
import Table from 'app/components/table'
import OrganisationChart from 'app/page'
import { EmployeesHierarchyData } from '../../src/types/employee/employee.type'
const mockHierarchy: EmployeesHierarchyData = {
  employeesHierarchy: [
    { id: 1, name: 'Darren', managerId: null, depth: 0 },
    { id: 2, name: 'Bobby', managerId: 1, depth: 1 },
  ],
  maxDepth: 1,
}

it('renders organisation chart unchanged', () => {
  const { container } = render(<OrganisationChart />)
  expect(container).toMatchSnapshot()
})

it('renders table unchanged', () => {
  const { container } = render(<Table hierarchy={mockHierarchy} />)
  expect(container).toMatchSnapshot()
})
