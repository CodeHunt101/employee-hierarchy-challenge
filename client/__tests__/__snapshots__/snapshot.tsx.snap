// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders organisation chart unchanged 1`] = `
<div>
  <h1>
    Failed to load data
  </h1>
</div>
`;

exports[`renders table unchanged 1`] = `
<div>
  <table
    class="table"
  >
    <tbody>
      <tr>
        <td>
          <b>
            Darren
          </b>
        </td>
        <td />
      </tr>
      <tr>
        <td />
        <td>
          <b>
            Bobby
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
